bindings/nodejs/lib/types/models/api/ITipsResponse.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import type { HexEncodedString } from '../../utils/hexEncodedTypes';
+/**
+ * Response from the tips endpoint.
+ */
+export interface ITipsResponse {
```

Comment:
When do we use an interface and when a class?
OutputResponse for example is a class, but also a response
Also regarding naming, do we want to use the `I` prefix for all interfaces?

Reply:
Yeah this is really confusing me as well. Is it really unthinkable that we don't use the I anywhere at all?

Reply:
Interfaces are easier to use because theres no class stuff `{}` can be `interface ITipsResponse {}`. However for easy construction without specifying a type, and having functions, we need a class. We could have a class and an interface for both, but that seems stupid. Ive left as much interface as possible, and transformed those that need it into a class. Unfortunately the json typescript deserialization also requires classes, not interfaces.

Reply:
@kwek20?

Reply:
yes, otherwise we couldnt transform it except manually transforming the output